java/clients/src/main/java/sleeper/status/report/CompactionJobStatusReport.java

```diff
@@ -0,0 +1,258 @@
+            String type = scanner.nextLine();
+            if ("".equals(type)) {
+                break;
+            }
+            if (!type.equalsIgnoreCase("d") && !type.equalsIgnoreCase("r")
+                    && !type.equalsIgnoreCase("u")) {
+                continue;
+            }
+            if (type.equalsIgnoreCase("d")) {
+                handleDetailedQuery(scanner);
+            } else if (type.equalsIgnoreCase("r")) {
+                handleRangeQuery(scanner);
+            } else {
+                handleUnfinishedQuery();
+            }
+        }
+    }
+
+    public void handleUnfinishedQuery() {
+        List<CompactionJobStatus> statusList = compactionJobStatusCollector.runUnfinishedQuery();
+        compactionJobStatusReporter.report(statusList, QueryType.UNFINISHED);
+    }
+
+    public void handleRangeQuery(Scanner scanner) {
+        Instant startTime = promptForStartRange(scanner);
+        Instant endTime = promptForEndRange(scanner);
+        handleRangeQuery(startTime, endTime);
+    }
+
+    public void handleRangeQuery(Instant startRange, Instant endRange) {
+        if (startRange.isAfter(endRange)) {
+            System.out.println("Start range provided is before end range. Using system defaults (past 4 hours)");
+            startRange = DEFAULT_RANGE_START;
+            endRange = DEFAULT_RANGE_END;
+        }
+        List<CompactionJobStatus> statusList = compactionJobStatusCollector.runRangeQuery(startRange, endRange);
+        compactionJobStatusReporter.report(statusList, QueryType.RANGE);
+    }
+
+    private Instant promptForStartRange(Scanner scanner) {
+        Instant startTime = DEFAULT_RANGE_START;
+        while (true) {
+            System.out.printf("Enter start range in format yyyy-MM-dd hh:mm:ss (default is %s):", DEFAULT_RANGE_START.toString());
+            String time = scanner.nextLine();
+            if ("".equals(time)) {
+                System.out.printf("Using default start range %s%n", DEFAULT_RANGE_START);
+                break;
+            }
+            try {
+                Date date = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss").parse(time);
+                startTime = date.toInstant();
+                break;
+            } catch (ParseException e) {
+                System.out.println("Error while parsing input string");
+            }
+        }
+        return startTime;
+    }
+
+    private Instant promptForEndRange(Scanner scanner) {
+        Instant endTime = DEFAULT_RANGE_END;
+
+        while (true) {
+            System.out.printf("Enter end range in format yyyy-MM-dd hh:mm:ss (default is %s):", DEFAULT_RANGE_END.toString());
+            String time = scanner.nextLine();
+            if ("".equals(time)) {
+                System.out.printf("Using default end range %s%n", DEFAULT_RANGE_END);
+                break;
+            }
+            try {
+                Date date = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss").parse(time);
```

Comment:
Maybe make new SimpleDateFormat("yyyy-MM-dd hh:mm:ss") a static variable, rather than creating the object every time?